### PR TITLE
Android: Parent module added, add BLE-components to timodule.xml

### PR DESCRIPTION
**Pull request by AppWerft**
issue with missing parent_module fixed,
device.getUuuids()  fixed.

### CODE REVIEW

File: android/timodule.xml [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
+						<action android:name="android.bluetooth.a2dp.profile.action.CONNECTION_STATE_CHANGED" />
+					</intent-filter>
+				</receiver>
+			</application>
 		</manifest>
 	</android>
-	<mobileweb>
-	</mobileweb>
+	<mobileweb></mobileweb>
 </ti:module>
```
Comment (hansemannn) [inline android/timodule.xml]: Can you put the changes of this file in an own PR? Then we could merge it directly.

Comment (AppWerft) [inline android/timodule.xml]: No idea how I can make a PR.

**Comment by m1ga** [comment]
Still crashing once a device is found:

```
12-31 13:08:34.276: D/BluetoothGatt(17158): connect() - callback: ti.bluetooth.BluetoothGattCallbackHandler@aa9422b
12-31 13:08:34.276: D/BluetoothGatt(17158): registerApp()
12-31 13:08:34.277: D/BluetoothGatt(17158): registerApp() - UUID=257c7000-00e1-451c-923e-1a7b0daae000
12-31 13:08:34.277: D/BtGatt.GattService(15886): registerClient() - UUID=257c7000-00e1-451c-923e-1a7b0daae000
12-31 13:08:34.278: D/BluetoothLeScanner(17158): stopScan()
12-31 13:08:34.279: E/ProxyFactory(17158): Failed to find class for ti.bluetooth.BluetoothDeviceProxy
12-31 13:08:34.279: E/ProxyFactory(17158): Unable to find class while creating V8 Proxy.
12-31 13:08:34.279: I/bt_att(15886): GATT_Register
12-31 13:08:34.279: D/bt_att(15886): UUID=[0x257c700000e1451c923e1a7b0daae000]
12-31 13:08:34.279: I/bt_att(15886): allocated gatt_if=6
12-31 13:08:34.280: I/bt_att(15886): GATT_StartIf gatt_if=6
12-31 13:08:34.280: I/bt_btif(15886): gatt_find_the_connected_bda start_idx=0
12-31 13:08:34.280: D/bt_att(15886): gatt_find_the_connected_bda start_idx=0
12-31 13:08:34.280: D/bt_att(15886): gatt_find_the_connected_bda found=0 found_idx=20
12-31 13:08:34.280: D/BtGatt.GattService(15886): onClientRegistered() - UUID=257c7000-00e1-451c-923e-1a7b0daae000, clientIf=6
12-31 13:08:34.280: D/BluetoothGatt(17158): onClientRegistered() - status=0 clientIf=6
12-31 13:08:34.280: D/A2dpService(15886): getA2DPService(): returning com.android.bluetooth.a2dp.A2dpService@b55b99a
12-31 13:08:34.281: I/A2dpService(15886): audio isMusicActive is false
12-31 13:08:34.282: D/BtGatt.GattService(15886): clientConnect() - address=00:00:00:00:00, isDirect=true
12-31 13:08:34.282: D/bt_btm(15886): BTM_SecAddBleDevice dev_type=0x2
12-31 13:08:34.282: D/bt_btm(15886): Device already exist
12-31 13:08:34.282: I/bt_btm(15886): BTM_InqDbRead: bd addr [f7c83ebb896e]
12-31 13:08:34.282: D/bt_btm(15886): InqDb  device_type =0x2  addr_type=0x1
12-31 13:08:34.282: I/bt_att(15886): GATT_Connect gatt_if=6
12-31 13:08:34.282: D/bt_att(15886): gatt_get_ch_state: ch_state=0
12-31 13:08:34.282: D/bt_att(15886): gatt_set_ch_state: old=0 new=2
12-31 13:08:34.282: I/bt_btm(15886): BTM_InqDbRead: bd addr [f7c83ebb896e]
12-31 13:08:34.282: D/bt_btm(15886): btm_find_dev_type - device_type = 2 addr_type = 1
12-31 13:08:34.282: I/bt_btm(15886): btm_find_or_alloc_dev
12-31 13:08:34.283: D/bt_att(15886): gatt_update_app_use_link_flag  is_add=1 chk_link=0
12-31 13:08:34.283: D/bt_att(15886): gatt_update_app_hold_link_status found=1[1-found] idx=0 gatt_if=6 is_add=1
12-31 13:08:34.283: D/bt_att(15886): gatt_get_ch_state: ch_state=2
12-31 13:08:34.283: I/bt_att(15886): GATT_GetConnIdIfConnected status=0
12-31 13:08:34.331: A/DEBUG(3621): *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
12-31 13:08:34.332: A/DEBUG(3621): Build fingerprint: 'htc/hiaeuhl_00401/htc_hiaeuhl:6.0.1/MMB29M/684509.81:user/release-keys'
12-31 13:08:34.332: A/DEBUG(3621): Revision: '0'
12-31 13:08:34.332: A/DEBUG(3621): ABI: 'arm'
12-31 13:08:34.332: A/DEBUG(3621): pid: 17158, tid: 17158, name: com.miga.ble  >>> com.miga.ble <<<
12-31 13:08:34.332: A/DEBUG(3621): signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0
12-31 13:08:34.357: A/DEBUG(3621):     r0 00000000  r1 00000000  r2 592d0b91  r3 00000000
12-31 13:08:34.357: A/DEBUG(3621):     r4 ab547b84  r5 00000000  r6 ab547b7c  r7 00000000
12-31 13:08:34.357: A/DEBUG(3621):     r8 ab547b54  r9 ab547b54  sl 00000000  fp 00000000
12-31 13:08:34.357: A/DEBUG(3621):     ip 00000000  sp fffd63b0  lr dd551a54  pc dd551ac4  cpsr 600f0010
12-31 13:08:34.362: A/DEBUG(3621): backtrace:
12-31 13:08:34.362: A/DEBUG(3621):     #00 pc 00746ac4  /data/app/com.miga.ble-1/lib/arm/libkroll-v8.so (_ZN2v88internal3Map24TransitionToDataPropertyENS0_6HandleIS1_EENS2_INS0_4NameEEENS2_INS0_6ObjectEEENS0_18PropertyAttributesENS6_14StoreFromKeyedE+232)
12-31 13:08:34.362: A/DEBUG(3621):     #01 pc 006eadc8  /data/app/com.miga.ble-1/lib/arm/libkroll-v8.so (_ZN2v88internal14LookupIterator31PrepareTransitionToDataPropertyENS0_6HandleINS0_8JSObjectEEENS2_INS0_6ObjectEEENS0_18PropertyAttributesENS5_14StoreFromKeyedE+180)
12-31 13:08:34.362: A/DEBUG(3621):     #02 pc 0072a5f4  /data/app/com.miga.ble-1/lib/arm/libkroll-v8.so (_ZN2v88internal6Object15AddDataPropertyEPNS0_14LookupIteratorENS0_6HandleIS1_EENS0_18PropertyAttributesENS1_11ShouldThrowENS1_14StoreFromKeyedE+660)
12-31 13:08:34.363: A/DEBUG(3621):     #03 pc 00736b8c  /data/app/com.miga.ble-1/lib/arm/libkroll-v8.so (_ZN2v88internal6Object11SetPropertyEPNS0_14LookupIteratorENS0_6HandleIS1_EENS0_12LanguageModeENS1_14StoreFromKeyedE+160)
12-31 13:08:34.363: A/DEBUG(3621):     #04 pc 0081d7f4  /data/app/com.miga.ble-1/lib/arm/libkroll-v8.so (_ZN2v88internal7Runtime17SetObjectPropertyEPNS0_7IsolateENS0_6HandleINS0_6ObjectEEES6_S6_NS0_12LanguageModeE+124)
12-31 13:08:34.363: A/DEBUG(3621):     #05 pc 003a6080  /data/app/com.miga.ble-1/lib/arm/libkroll-v8.so (_ZN2v86Object3SetENS_5LocalINS_7ContextEEENS1_INS_5ValueEEES5_+240)
12-31 13:08:34.363: A/DEBUG(3621):     #06 pc 003a619c  /data/app/com.miga.ble-1/lib/arm/libkroll-v8.so (_ZN2v86Object3SetENS_5LocalINS_5ValueEEES3_+60)
12-31 13:08:34.363: A/DEBUG(3621):     #07 pc 002c7695  /data/app/com.miga.ble-1/lib/arm/libkroll-v8.so (_ZN8titanium13TypeConverter20javaHashMapToJsValueEPN2v87IsolateEP7_JNIEnvP8_jobject+196)
12-31 13:08:34.363: A/DEBUG(3621):     #08 pc 002c8477  /data/app/com.miga.ble-1/lib/arm/libkroll-v8.so (Java_org_appcelerator_kroll_runtime_v8_V8Object_nativeFireEvent+202)
12-31 13:08:34.363: A/DEBUG(3621):     #09 pc 006283b3  /data/app/com.miga.ble-1/oat/arm/base.odex (offset 0x439000)
12-31 13:08:34.676: A/DEBUG(3621): Tombstone written to: /data/tombstones/tombstone_04
12-31 13:08:34.676: E/DEBUG(3621): AM write failed: Broken pipe
```

File: android/src/ti/bluetooth/TiBluetoothModule.java [hunk 2]
```diff
@@ -8,301 +8,254 @@
...
-        if (btAdapter != null) {
-            btScanner.stopScan(scanCallback);
-            isScanning = false;
-        }
-    }
+	public interface ConnectionCallback {
+		void onConnectionStateChange(BluetoothDevice device, int newState);
+	}
+
+	public static final String LCAT = "BLE🔵";
```
Comment (m1ga) [inline android/src/ti/bluetooth/TiBluetoothModule.java]: Please remove the blue dot:
ble

Comment (hansemannn) [inline android/src/ti/bluetooth/TiBluetoothModule.java]: Shouldn't the tag be configurable anyway?

Comment (m1ga) [inline android/src/ti/bluetooth/TiBluetoothModule.java]: It's just the debug log tag like in https://github.com/appcelerator-modules/ti.map/blob/master/android/src/ti/map/TiUIMapView.java#L58

**Comment by AppWerft** [comment]
What I can/should do?

**Comment by m1ga** [comment]
Most of the changes are already included (without the crashing part :) ), so I think I'll close this PR and check the missing parts and include them.

**Comment by AppWerft** [comment]
Good idea.

**Pull request closed without merging**